### PR TITLE
fix: cherry-picks to standard branches now supported

### PR DESCRIPTION
**Pull request by oleorhagen**
Previously cherry-picks were only supported in release branches matching some
release version, like `1.2.x`.

Now, this functionality is extended to the standard branchnames, like:

* master
* hosted
* staging
* production

Changelog: None
Signed-off-by: Ole Petter <ole.orhagen@northern.tech>

### CODE REVIEW

**Comment by oleorhagen** [comment]
I'm guessing, we really could open this up to any branch name?

What do you think?

**Comment by lluiscampos** [comment]
> I'm guessing, we really could open this up to any branch name?
> 
> What do you think?

yes, why not?

**Review by lluiscampos — APPROVED** [review]
Looks good

But I think we could remove the check all together. We usually know what we are doing when asking for a cherry-pick, and we still will review every PR before merge. So I don't see any risk in allowing any arbitrary branch.

**Review by lluiscampos — APPROVED** [review]

**Comment by lluiscampos** [comment]
@oleorhagen I took the liberty of fixing the unit tests and force-push to your branch. I want to merge this to unblock another PR.

Side note, the three test cases of `TestSuggestCherryPicks` are really the same now, should we unify in one?

**Comment by oleorhagen** [comment]
> @oleorhagen I took the liberty of fixing the unit tests and force-push to your branch. I want to merge this to unblock another PR.
> 
> Side note, the three test cases of `TestSuggestCherryPicks` are really the same now, should we unify in one?

Alright, nice!

I was actually planning on waiting for your release, but same matter. Or rather, even better actually ;)

**Comment by oleorhagen** [comment]
> Side note, the three test cases of TestSuggestCherryPicks are really the same now, should we unify in one?

Sure

**Pull request merged**
